Tasks/AzureRmWebAppDeployment/AzureRmWebAppDeployment.ts

```diff
@@ -0,0 +1,235 @@
+            await DeployUsingKuduDeploy(webDeployPkg, azureWebAppDetails, publishingProfile, virtualApplication, isFolderBasedDeployment);
+        }
+    } catch (error) {
+        tl.setResult(tl.TaskResult.Failed, error);
+    }
+}
+
+/**
+ * Executes Web Deploy command
+ * 
+ * @param   webDeployPkg                   Web deploy package
+ * @param   webAppName                      web App Name
+ * @param   publishingProfile               Azure RM Connection Details
+ * @param   removeAdditionalFilesFlag       Flag to set DoNotDeleteRule rule
+ * @param   excludeFilesFromAppDataFlag     Flag to prevent App Data from publishing
+ * @param   takeAppOfflineFlag              Flag to enable AppOffline rule
+ * @param   virtualApplication              Virtual Application Name
+ * @param   setParametersFile               Set Parameter File path
+ * @param   additionalArguments             Arguments provided by user
+ * 
+ */
+async function DeployUsingMSDeploy(webDeployPkg, webAppName, publishingProfile, removeAdditionalFilesFlag, 
+        excludeFilesFromAppDataFlag, takeAppOfflineFlag, virtualApplication, setParametersFile, additionalArguments, isFolderBasedDeployment) {
+
+    var isParamFilePresentInPackage = isFolderBasedDeployment ? false : await msDeployUtility.containsParamFile(webDeployPkg);
+    setParametersFile = getSetParamFilePath(setParametersFile);
+    var msDeployPath = await msDeployUtility.getMSDeployFullPath();
+    var msDeployCmdArgs = msDeployUtility.getMSDeployCmdArgs(webDeployPkg, webAppName, publishingProfile, removeAdditionalFilesFlag,
+        excludeFilesFromAppDataFlag, takeAppOfflineFlag, virtualApplication, setParametersFile, additionalArguments, isParamFilePresentInPackage, isFolderBasedDeployment);
+
+    var isDeploymentSuccess = true;
+    var deploymentError = null;
+    try {
+
+        var msDeployBatchFile = tl.getVariable('System.DefaultWorkingDirectory') + '\\' + 'msDeployCommand.bat';
+        var silentCommand = '@echo off \n';
+        var msDeployCommand = '"' + msDeployPath + '" ' + msDeployCmdArgs;
+        var batchCommand = silentCommand + msDeployCommand;
+
+        tl.writeFile(msDeployBatchFile, batchCommand);
+        tl._writeLine(tl.loc("Runningcommand", msDeployCommand));
+        await tl.exec("cmd", ['/C', msDeployBatchFile], <any> {failOnStdErr: true});
+        tl._writeLine(tl.loc('WebappsuccessfullypublishedatUrl0', publishingProfile.destinationAppUrl));
+    }
+    catch(error) {
+        tl.error(tl.loc('Failedtodeploywebsite'));
+        isDeploymentSuccess = false;
+        deploymentError = error;
+    }
+
+    try {
+        tl._writeLine(await azureRmUtil.updateDeploymentStatus(publishingProfile, isDeploymentSuccess));
+    }
+    catch(error) {
+        tl.warning(error);
+    }
+    finally {
+        if(!isDeploymentSuccess) {
+            throw Error(deploymentError);
```

Comment:
Move this code out of finally block as this not at all related to previous try/catch